### DISCUSSION TITLE
Compatibility with non American QWERTY Keyboards (eg: AZERTY)

### DISCUSSION THREAD

**Discussion by notexplosive** [MonoGame/MonoGame#8321] [Inputs]
Hiya! 👋 First time posting here.

It seems that the only way to check a particular key is with the `Keys` enum.

If my game supports WASD, how does that translate to a French ([AZERTY](https://en.wikipedia.org/wiki/AZERTY)) keyboard. Does the WASD magically map to ZQSD (the same key placements, but different letters) or does it map to the "WASD" keys on their keyboard (which are all on the wrong places).

Likewise, is there a way to tell what kind of keyboard layout the player is using? (ie: can I hint "Use ZQSD to move" instead of "Use WASD to move")

I'm curious how other people solved this.

**Comment by ThomasFOG** [comment]
Hi!

> Does the WASD magically map to ZQSD (the same key placements, but different letters) or does it map to the "WASD" keys on their keyboard (which are all on the wrong places).

If someone types "A" on an AZERTY or a QWERTY keyboard, MonoGame will detect A no matter where that key is physically on the keyboard.

> Likewise, is there a way to tell what kind of keyboard layout the player is using?

Not with MonoGame. You can use system queries but that's not ideal.

> I'm curious how other people solved this.

I'm French and from experience, AZERTY will be very rare and you can simply assume that all your players are on QWERT-. That being said, it is strongly advised to not hard-code your prompts and to have key remapping and your prompts properly showing what's mapped. Unless you want negative reviews on Steam.

**Comment by Apostolique** [comment]
There are more discussions for this in #6617.
